src/components/hooks/useGetRecipePrice.tsx

```diff
@@ -0,0 +1,103 @@
+import { IngredientData, NewRecipe, Product } from "../types.ts";
+import { useGetIngredientIds } from "./useGetIngredientsIds.tsx";
+
+export const useGetRecipePrice = (recipe: NewRecipe) => {
```

Comment:
Tohe je monstrum funkce. Zajimalo by me, jestli by ta vnitrni implementace sla napsat trochu elegantneji, muzem si to zkusit jako takovou challenge, ale prvne pro to napisem test. Muzeme to probrat treba na dalsim callu, super kandidat! 🚀 Dobre jsi se s tim poprala, predpokladam, ze trochu potu, krve a slz padlo :D